src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,14 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            Debug.Assert(_property.IsExpressionBodied == false, "Cannot have accessors in expression bodied lightweight properties");
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null);
```

Comment:
I'll refactor that in the next PR (e.g. when adding tests)

---
In reply to: [77268250](https://github.com/dotnet/roslyn/pull/13543#discussion_r77268250) [](ancestors = 77268250)